### PR TITLE
added agent groups feature test

### PR DESCRIPTION
**Pull request by manrodrigues**

### CODE REVIEW

**Review by stefano-daitan — COMMENTED** [review]
Good job, I only have a few comments related to wording.

File: python-test/features/steps/local_agent.py [hunk 1]
```diff
@@ -24,9 +25,18 @@ def run_local_agent_container(context):
...
 def check_agent_log(context, text_to_match):
-    logs = get_orb_agent_logs(context.container_id)
-    contains_text = check_logs_contains_message(logs, text_to_match)
-    assert_that(contains_text, is_(True), 'Message "' + text_to_match + '" was not found in the agent logs!')
+    time_waiting = 0
+    sleep_time = 0.5
+    timeout = 10
+    while time_waiting < timeout:
+        logs = get_orb_agent_logs(context.container_id)
+        contains_text = check_logs_contains_message(logs, text_to_match)
```
[the remark this answers is not in the document]
Comment (stefano-daitan) [reply python-test/features/steps/local_agent.py:33]: or check_log_contains_message

**Review by weyrick — DISMISSED** [review]

File: python-test/features/steps/control_plane_agents.py [hunk 2]
```diff
@@ -89,16 +87,18 @@ def list_agents(token):
     return agents_as_json['agents']
 
 
-def delete_agents(token, list_of_agents):
+def delete_agents(token, list_of_agents, start_with):
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_agents.py]: ok.. I created a new function to filter elements.. once i can reuse the same function in many scenarios, I put it on utils.py

File: python-test/features/steps/control_plane_agent_groups.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+from behave import given, when, then
+from hamcrest import *
+from test_config import TestConfig
+import requests
+from users import get_auth_token
+
+configs = TestConfig.configs()
+agent_group_name = 'test_group_name_' + random_string()
+agent_group_description = "This is an agent group"
+agent_name = "test_agent_name_" + random_string(4)
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_agent_groups.py]: ok.. on control_plane_agents.py I created a variable called "agent_name_prefix" and on control_plane_agent_groups.py I created one called agent_group_name_prefix.

File: python-test/features/steps/control_plane_agents.py [hunk 4]
```diff
@@ -135,3 +135,18 @@ def create_agent(token, name, tag_key, tag_value):
...
+    Remove all agents starting with 'test_agent_name_' from the orb
+
+    :param context: Behave class that contains contextual information during the running of tests.
+    """
+    context.execute_steps('''
+    Given that the user is logged in
+    ''')
+    token = context.token
+    agents_list = list_agents(token)
+    delete_agents(token, agents_list, 'test_agent_name_')
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_agents.py]: good idea.

File: python-test/features/steps/control_plane_agent_groups.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+
+@then('cleanup agent group')
+def clean_agent_groups(context):
+    """
+    Remove all agent groups starting with 'test_group_' from the orb
+
+    :param context: Behave object that contains contextual information during the running of tests.
+    """
+    get_auth_token(context)
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_agent_groups.py]: Once we are running this on "after_scenario" scope, context is empty so we need to run the auth scenario. I put this on enviromentals.py in order to guarantee that we gonna do that just once.

File: python-test/features/steps/control_plane_agent_groups.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+agent_tag_key = "test_tag_key_" + random_string(4)
+agent_tag_value = "test_tag_value_" + random_string(4)
+
+
+@given("that an agent already exists and is {status}")
+def check_if_agents_exist(context, status):
+    agent = create_agent(context.token, agent_name, agent_tag_key, agent_tag_value)
+    context.agent = agent
+    token = context.token
+    run_local_agent_container(context)
```
[the remark this answers is not in the document]
Comment (manrodrigues) [reply python-test/features/steps/control_plane_agent_groups.py:24]: Yes. You are right. We can solve this provisionally by cleanup the containers after all features. If in the future we have to have 2 runs on the same feature, we can change the container name, or get another strategy

[the remark this answers is not in the document]
Comment (stefano-daitan) [reply python-test/features/steps/control_plane_agent_groups.py:24]: that should be sufficient for now

**Review by stefano-daitan — APPROVED** [review]

**Pull request merged**